flowmachine/flowmachine/utils.py

```diff
@@ -377,9 +373,28 @@ def sort_recursively(d):
             d_new[key] = sort_recursively(d[key])
         return d_new
     elif isinstance(d, list):
-        try:
-            return sorted(d, key=to_nested_list)
-        except:
-            breakpoint()
+        return sorted(d, key=to_nested_list)
     else:
         return d
+
+
+def print_dependency_tree(query_obj, level=0, stream=sys.stdout):
```

Comment:
Can we combine this with `Query.dependency_graph`? Ideally by removing `Query.dependency_graph` in favour of this ;P

Reply:
How do you imagine them being combined? I actually like both options - being able to print a dependency tree in plain-text format, and also to plot it as a full graph. Should I just rip the `dependency_graph()` function out of the `Query` class and put it here in `utils.py`, or do you have something else in mind?

Reply:
It isn't so much the two options, but the two options with overlapping functionality being in two different places I think.

Reply:
Would you prefer both of them to live as methods on the `Query` class, or rather as separate helper functions in `utils.py`?

Reply:
Definitely in utils. `Query` is already way too bloated.